axiom/tests/test_log_events.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nr_axiom.h"
+
+#include <stddef.h>
+
+#include "nr_log_events.h"
+#include "util_random.h"
+#include "util_strings.h"
+
+#include "tlib_main.h"
+
+#define LOG_LEVEL "INFO"
+#define LOG_MESSAGE "this is a test log error message"
+#define LOG_MESSAGE_0 LOG_MESSAGE " 0"
+#define LOG_MESSAGE_1 LOG_MESSAGE " 1"
+#define LOG_MESSAGE_2 LOG_MESSAGE " 2"
+#define LOG_MESSAGE_3 LOG_MESSAGE " 3"
+#define LOG_TIMESTAMP 12345
+
+static nr_log_event_t* create_sample_event(const char* message) {
+  nr_log_event_t* e = nr_log_event_create();
+  nr_log_event_set_log_level(e, LOG_LEVEL);
+  nr_log_event_set_message(e, message);
+  nr_log_event_set_timestamp(e, LOG_TIMESTAMP * NR_TIME_DIVISOR_MS);
+  return e;
+}
+
+static void test_events_success(void) {
+  /* Normal operation */
+  nr_analytics_events_t* events = NULL;
+  nr_log_event_t* e = NULL;
+  nr_random_t* rnd = nr_random_create_from_seed(12345);
+  const char* json;
+  const char* expected;
+
+  events = nr_log_events_create(10);
+  tlib_fail_if_null("events created", events);
```

Comment:
I do not think this is needed because:
1. `nr_analytics_events_t` is an opaque type that does not allow accessing individual fields.
2. There already exist [axiom/tests/test_analytics_events.c](https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/tests/test_analytics_events.c) that tests analytic events API, which is the underlying and re-used implementation of log events API.

Reply:
That seems reasonable.